service/actions/helpers.js

```diff
@@ -8,6 +8,8 @@ const HTTP_UNAUTHORIZED = 401;
 export async function forceLogoutIfNecessary(err, dispatch) {
     if (err.status_code === HTTP_UNAUTHORIZED && err.url.indexOf('/login') === -1) {
         await Client.logout();
+
+        // TODO: we need to find a way to clear the storage too
```

Comment:
That's a very good question. I wonder if we need to let the app layer pass in some sort of callback to happen on LOGOUT. It's also kind of weird how I have the navigation store listening for this LOGOUT action as well.